numba/tests/test_array_methods.py

```diff
@@ -484,6 +487,14 @@ def check(buf):
             expected = pyfunc(buf)
             self.memory_leak_setup()
             got = run(buf)
+            # TODO: why is run() producing a frame that holds reference if the
+            # dtype is a str literal?
+            if 'dtype_str' in self.id():
+                import gc
+                frames = gc.get_referrers(buf)
+                if frames:
+                    del frames[0]
+                gc.collect()
```

Comment:
@sklam this is the weird frame holding ref problem.

Reply:
the frame is from `run`? 
```python
<frame at 0x614314410, file 'numba/tests/test_array_methods.py', line 484, code run>
```

Reply:
yes, I saw the same, didn't debug much further

Reply:
Just tried to separate the compiling and running. It's certainly from the compiling; i.e. `self.ccache.compile(pyfunc, (typeof(buf),))`

Reply:
it's the try...except in `support.py:126`

Reply:
so with this patch:

```diff
diff --git a/numba/tests/support.py b/numba/tests/support.py
index ae2430f1a..f9c05a7fe 100644
--- a/numba/tests/support.py
+++ b/numba/tests/support.py
@@ -123,9 +123,9 @@ class CompilationCache(object):
         from numba.core.registry import cpu_target

         cache_key = (func, args, return_type, flags)
-        try:
+        if cache_key in self.cr_cache:
             cr = self.cr_cache[cache_key]
-        except KeyError:
+        else:
             # Register the contexts in case for nested @jit or @overload calls
             # (same as compile_isolated())
             with cpu_target.nested_context(self.typingctx, self.targetctx):
```

Reply:
ah, that would explain it, thanks!

Reply:
3be2086 implements